src/backend/src/services/tasks.services.ts

```diff
@@ -51,7 +55,18 @@ export default class TasksService {
     const isProjectLeadOrManager =
       createdBy.userId === requestedWbsElement.projectLeadId || createdBy.userId === requestedWbsElement.projectManagerId;
 
-    if (!isLeadership(createdBy.role) && !isProjectLeadOrManager && !teams.some((team) => isUserOnTeam(team, createdBy))) {
+    const curWorkPackages = requestedWbsElement.project?.workPackages;
+
+    const isWorkPackageLead = curWorkPackages?.some((workPackage) => {
+      return workPackage.wbsElement.projectLeadId === createdBy.userId;
+    });
+
+    if (
+      !isLeadership(createdBy.role) &&
+      !isProjectLeadOrManager &&
+      isWorkPackageLead &&
```

Comment:
ooh minor point here! Since `isWorkPackageLead `variable returns true when at least one of workpackage lead id matches to createdBy.userId, we want only to throw an error when `isWorkPackageLead `returns false.